### PR TITLE
storage: store the partition name in device attrs for partitions

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: tests/unit_tests/pyanaconda_tests/modules/storage/test_module_device_tree.py [hunk 1]
```diff
@@ -341,6 +342,23 @@ def test_get_zfcp_device_data(self):
...
+        sda1._parted_partition = Mock(**{'type': PARTITION_NORMAL,
+                                         'getLength.return_value': int(sda1.size),
+                                         'getFlag.return_value': 0,
+                                         'name': "Microsoft reserved partition",
+                                         'number': 1})
+
+        self._add_device(sda1)
+
+        data = self.interface.GetDeviceData("dev1")
+        # FIXME: assert data['parted-partition-name'] is Microsoft..
```
Comment (vojtechtrefny) [inline tests/unit_tests/pyanaconda_tests/modules/storage/test_module_device_tree.py:360]: `name` is an argument for the Mock constructor so you need to use `configure_mock` to set it after creating the `Mock` instance, see https://python.readthedocs.io/en/latest/library/unittest.mock.html#mock-names-and-the-name-attribute

Comment (KKoukiou) [reply tests/unit_tests/pyanaconda_tests/modules/storage/test_module_device_tree.py]: Thank you

File: pyanaconda/modules/common/structures/storage.py [hunk 2]
```diff
@@ -196,6 +196,9 @@ def attrs(self) -> Dict[Str, Str]:
             hba-id
             path-id
 
+        Attributes for partitions:
+            parted-partition-name
```
Comment (jkonecny12) [inline pyanaconda/modules/common/structures/storage.py:200]: The `parted` here is the name of the `parted` project which is used by blivet. I don't think it should be part of the Anaconda API, it's something which might be confusing in the future.

The best candidate so far seems to me `partition-label` but in general I guess we need to document it to be able to understand what it is.

Comment (KKoukiou) [reply pyanaconda/modules/common/structures/storage.py]: Renamed.

**Pull request merged**